tests/integration/ha_tests/application_charm/src/charm.py

```diff
@@ -120,36 +123,24 @@ def _stop_continuous_writes(self) -> Optional[int]:
         if not self.app_peer_data.get(PROC_PID_KEY):
             return None
 
-        # Send a SIGKILL to the process and wait for the process to exit
-        proc = subprocess.Popen(["pkill", "--signal", "SIGKILL", "-f", "src/continuous_writes.py"])
-        proc.communicate()
+        # Send a SIGTERM to the process and wait for the process to exit
+        os.kill(int(self.app_peer_data[PROC_PID_KEY]), signal.SIGTERM)
 
         del self.app_peer_data[PROC_PID_KEY]
 
-        # Query and return the max value inserted in the database
-        # (else -1 if unable to query)
+        # read the last written_value
         try:
             for attempt in Retrying(stop=stop_after_delay(60), wait=wait_fixed(5)):
                 with attempt:
-                    last_written_value = self._max_written_value()
+                    with open(LAST_WRITTEN_FILE, "r") as fd:
```

Comment:
It may take a bit for the `continuous_writes` loop to exit, so I recycled the retrying.